### ISSUE TITLE
In raw mode on Windows, Deno.stdin.readSync drops LF character from CRLF

### ISSUE THREAD

**Issue by Rummskartoffel** [denoland/deno#21837]
Version: Deno 1.39.2

To reproduce:
Save the following code to `bug.ts`:
```typescript
let a = new Uint8Array(2), b = new Uint8Array(2)
Deno.stdin.readSync(a)
Deno.stdin.setRaw(true)
Deno.stdin.readSync(b)
console.log(a)
console.log(b)
```
In a terminal, run:
```
deno run bug.ts
```
Hit <kbd>Return</kbd> twice.

---
Actual output:
```
Uint8Array(2) [ 13, 10 ]
Uint8Array(2) [ 13, 0 ]
```
The non-raw read returned \<CR>\<LF>, the raw read returned only \<CR>.

Expected output:
```
Uint8Array(2) [ 13, 10 ]
Uint8Array(2) [ 13, 10 ]
```
Both reads should return \<CR>\<LF>.

**Comment by nhrones** [comment]
There currently is a problem with the Fresh cli that may be related.  When asked for one of the options like twind, hitting `enter` to accept the default does not work.  The enter does create a newline in the terminal, but the default value is not used, and the dialog seems to be stuck and never completes.

**Comment by kt3k** [comment]
On mac, the above script prints:
```
Uint8Array(2) [ 10, 0 ]
Uint8Array(2) [ 13, 0 ]
```

I'm not sure this is a bug. It seems that the Enter key consistently becomes `CR` in different platforms?

Can anyone test on linux?

(Note: downgrading to 1.39.1 doesn't make any change)

**Comment by codesculpture** [comment]
Yes, there has been problem with windows especially with Powershell, a related issue is found on `dart`
https://github.com/dart-lang/sdk/issues/54588

**Comment by codesculpture** [comment]
While i try to fix this on dart, @mraleph noted that this is issue from https://github.com/microsoft/terminal/issues/16223 and seems it was landed but no idea in which windows sdk version it was released. I think it can be closed @kt3k @Rummskartoffel

**Comment by kt3k** [comment]
Can anyone on windows test the original [example](https://github.com/denoland/deno/issues/21837#issue-2069211547) with the latest Windows Terminal?

**Comment by codesculpture** [comment]
The above output becomes 
```
```

**Comment by Rummskartoffel** [comment]
@codesculpture, my results (now on Deno 1.40.4, WT 1.18.10301.0) are still as reported. I only get your results when I run the test program multiple times in a row, so the raw flag is still set from the previous run.

**Comment by codesculpture** [comment]
Sorry for the confusion i actually tested on v1.18.2 can u try that @Rummskartoffel

**Comment by codesculpture** [comment]
@Rummskartoffel any update on this, have u tested on win terminal v1.18.2

**Comment by Rummskartoffel** [comment]
@codesculpture, I'm not sure which version you mean by "v1.18.2" – that's not a version number listed at [microsoft/terminal](https://github.com/microsoft/terminal/releases) –, but I've now updated to WT 1.19.10302.0 and the behaviour is unchanged.
